### PR TITLE
ci: Generate SBOM

### LINKED ISSUE #159: Generate SBOMs for .NET components
We have SBOMs currently for Java and Go. We could use them here as well. I recommend this utility: https://github.com/marketplace/actions/cyclonedx-net-generate-sbom (we're using the clyclonedx format elsewhere and it's popular).

Definition of done:

- SBOMs generated and attached to release artifact in GH, or otherwise made publicly available (for every release)
- only includes dependencies of module in question (not of repo)

### PR DESCRIPTION
**Pull request by jenshenneberg**
## This PR
Generates Software Bill of Materials (SBOM) as described in #159. Once https://github.com/NuGet/Home/issues/12497 is implemented, the SBOM file(s) should be embedded in the published nuget packages. Until then, I've added the SBOM as an asset under the release. 

### Known issue
The SBOM file lists the dependences for all target frameworks combined. Once the above [NuGet ](https://github.com/NuGet/Home/issues/12497)issue is implemented, it should be changed, so there is one sbom created for each target framework with only the applicable references included.

### Related Issues
Fixes #159

### How to test
Unfortunately, this is somewhat cumbersome to test, as the logic in question only kicks in upon a release from the main branch. I've tested it myself this way:
- Create new fork of this repo
- Merge this branch to main in the new repo
- Create a release in the new repo

### CODE REVIEW

**Comment by toddbaert** [comment]
Wow thanks @jenshenneberg ! I will review this carefully tomorrow.

File: .github/workflows/release.yml
Comment (beeme1mr) [inline .github/workflows/release.yml:55]: It may be worth enabling [continue-on-error](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error) until we've seen this job run successfully.

**Review by beeme1mr — APPROVED** [review]
Looks good but we may want to preview the CI from failing until it has run successfully.

File: .github/workflows/release.yml
Comment (jenshenneberg) [reply .github/workflows/release.yml:55]: It should not block the release even if it fails, but it's a good point regardless to keep the noise down in case of a failure. I'll add your suggestion.

Comment (jenshenneberg) [reply .github/workflows/release.yml:56]: I've added `continue-on-error` now 😃

**Review by beeme1mr — APPROVED** [review]
Awesome, thank you! 🎉

**Review by toddbaert — APPROVED** [review]
Ya this looks good to me, thanks for also extracting the conditions.

I'll merge tomorrow unless I hear objections from maintainers.

**Review by benjiro — APPROVED** [review]

**Review by askpt — APPROVED** [review]

**Pull request merged**
